airflow/www/views.py

```diff
@@ -3574,23 +3574,28 @@ def datasets_summary(self):
 
             count_query = session.query(func.count(DatasetModel.id))
 
+            has_event_filters = bool(updated_before or updated_after)
+
             query = (
                 session.query(
                     DatasetModel.id,
                     DatasetModel.uri,
                     func.max(DatasetEvent.timestamp).label("last_dataset_update"),
-                    func.count(distinct(DatasetEvent.id)).label("total_updates"),
+                    func.count(func.sum(func.case(DatasetEvent.id.is_not(None), 1, else_=0))).label(
```

Comment:
when no event filters, we do a left join, so event may be null, though dataset exists, so total updates should be _zero_, hence this case when